x/launch/keeper/request_test.go

```diff
@@ -76,3 +93,111 @@ func TestRequestCount(t *testing.T) {
+		}, {
+			name:    "test not found genesis AccountRemoval content",
+			request: *sample.RequestWithContent(chainID, contents[1]),
+			err: sdkerrors.Wrapf(types.ErrAccountNotFound,
+				"account %s for chain %s not found", genesisAcc, chainID),
+		}, {
+			name:    "test VestedAccount content",
+			request: *sample.RequestWithContent(chainID, contents[2]),
+		}, {
+			name:    "test duplicated VestedAccount content",
+			request: *sample.RequestWithContent(chainID, contents[2]),
+			err: sdkerrors.Wrapf(types.ErrAccountAlreadyExist,
+				"account %s for chain %s already exist", vestedAcc, chainID),
+		}, {
+			name:    "test vested AccountRemoval content",
+			request: *sample.RequestWithContent(chainID, contents[3]),
+		}, {
+			name:    "test not found vested AccountRemoval content",
+			request: *sample.RequestWithContent(chainID, contents[3]),
+			err: sdkerrors.Wrapf(types.ErrAccountNotFound,
+				"account %s for chain %s not found", vestedAcc, chainID),
+		}, {
+			name:    "test GenesisValidator content",
+			request: *sample.RequestWithContent(chainID, contents[4]),
+		}, {
+			name:    "test duplicated GenesisValidator content",
+			request: *sample.RequestWithContent(chainID, contents[4]),
+			err: sdkerrors.Wrapf(types.ErrValidatorAlreadyExist,
+				"genesis validator %s for chain %s already exist", validatorAcc, chainID),
+		}, {
+			name:    "test ValidatorRemoval content",
+			request: *sample.RequestWithContent(chainID, contents[5]),
+		}, {
+			name:    "test not found ValidatorRemoval content",
+			request: *sample.RequestWithContent(chainID, contents[5]),
+			err: sdkerrors.Wrapf(types.ErrValidatorNotFound,
+				"genesis validator %s for chain %s not found", validatorAcc, chainID),
+		}, {
+			name:    "invalid request",
+			request: *sample.RequestWithContent(chainID, invalidContent),
+			err: spnerrors.Critical(
+				"unknown request content type"),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := applyRequest(sdkCtx, *k, chainID, tt.request)
+			if tt.err != nil {
+				require.Error(t, err)
+				require.Equal(t, tt.err.Error(), err.Error())
+				return
+			}
+			require.NoError(t, err)
+
+			var content types.RequestContent
+			cdc := codectypes.NewInterfaceRegistry()
+			types.RegisterInterfaces(cdc)
+			err = cdc.UnpackAny(tt.request.Content, &content)
+			require.NoError(t, err)
+
+			switch c := content.(type) {
+			case *types.GenesisAccount:
+				_, found := k.GetGenesisAccount(sdkCtx, chainID, c.Address)
+				require.True(t, found, "genesis account not found")
+			case *types.VestedAccount:
+				_, found := k.GetVestedAccount(sdkCtx, chainID, c.Address)
+				require.True(t, found, "vested account not found")
+			case *types.AccountRemoval:
+				_, foundGenesis := k.GetGenesisAccount(sdkCtx, chainID, c.Address)
+				_, foundVested := k.GetVestedAccount(sdkCtx, chainID, c.Address)
+				require.False(t, foundGenesis && foundVested, "account not removed")
```

Comment:
```suggestion
				require.False(t, foundGenesis || foundVested, "account not removed")
```

Should be removed from both places. Here the assertion would succeed is one exists